docs/tutorial/visualization.rst

```diff
@@ -0,0 +1,420 @@
+
+    artist = BoxArtist(box)
+    a = artist.draw()
+
+.. raw:: html
+
+    </div>
+    <div class="tab-pane" id="object-artist-blender">
+
+.. code-block:: python
+
+    from compas.geometry import Frame, Box
+    from compas_blender.artists import BoxArtist
+
+    box = Box(Frame.worldXY(), 1, 1, 1)
+
+    artist = BoxArtist(box)
+    artist.draw()
+
+.. raw:: html
+
+    </div>
+    </div>
+    </div>
+    </div>
+
+
+Base Artist
+===========
+
+The base artist class for all artists is :py:class:`compas.artists.Artist`.
+This base class can be used to create the correct artist type for any of the COMPAS object types.
+The type of artist that is created depends on the combination of the object type and the context in which the code is executed: Rhino, GH, or Blender.
+This eliminates the need for more specific artist imports and allows for creating general scripts that can be run in Rhino, GH, and Blender without further modifications.
+
+.. code-block:: python
+
+    from compas.artists import Artist
+    from compas.geometry import Frame, Box
+
+    box = Box(Frame.worldXY(), 1, 1, 1)
+
+    artist = Artist(box)
+    artist.draw()
+
+
+When running this code snippet in Rhino (using ``RunPythonScript``) the created artist will be of type :class:`compas_rhino.artists.BoxArtist`.
+
+.. code-block:: python
+
+    print(type(artist))
+    # <class 'compas_rhino.artists.boxartist.BoxArtist'>
+
+
+The base artist detected that the current execution context is Rhino,
+and that the object passed to the constructor is a :py:class:`compas.geometry.Box`.
+
+In Blender, we will get a :class:`compas_blender.artists.BoxArtist`.
+
+.. code-block:: python
+
+    print(type(artist))
+    # <class 'compas_blender.artists.boxartist.BoxArtist'>
+
+
+In Grasshopper, the snippet can be executed from within a GH User Component, with a small modification.
+The output of the draw function has to be assigned to the output variable of the component (e.g. output variable ``a``).
+
+.. code-block:: python
+
+    from compas.artists import Artist
+    from compas.geometry import Frame, Box
+ 
+    box = Box(Frame.worldXY(), 1, 1, 1)
+
+    artist = Artist(box)
+    a = artist.draw()
+
+
+.. code-block:: python
+
+    print(type(artist))
+    # <class 'compas_ghpython.artists.boxartist.BoxArtist'>
+
+
+Colors
+======
+
+Colors are in the form of RGB tuples.
+In Rhino and Grasshopper the components have to be specified in integer format with values between 0 and 255.
+In Blender, colors have RGB components in float format with values between 0 and 1.
+Values beyond 1 will cause the object to emit light with an intensity related to the provided number.
+Therefore, the same colors can be used in Rhino, Grasshopper, and Blender, if they are in 0-255 integer format.
+
+.. code-block:: python
+
+    from compas.geometry import Frame, Box
+    from compas.artists import Artist
+
+    box = Box(Frame.worldXY(), 1, 1, 1)
+
+    Artist(box).draw(color=(0, 255, 0))
```

Comment:
Why don't we normalize the color definition to be either 0-255 or 0-1 for everything? It makes little sense to have this unmapped, being so easy to simplify for the user.

Reply:
that would be great but requires a bit of infrastructure, since all Rhino related color definitions would have to be converted to 0-255. perhaps through color plugins?

Reply:
yep, could be a color mapping plugin or something like that...